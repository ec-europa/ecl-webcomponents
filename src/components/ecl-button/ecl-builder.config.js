const path = require('path');
const pkg = require('./package.json');

const nodeModules = path.resolve(__dirname, '../../../node_modules');
const outputFolder = './build';

// SCSS includePaths
const includePaths = [nodeModules];

const banner = `${pkg.name} - ${pkg.version} Built on ${new Date().toISOString()}`;

module.exports = {
  styles: [
    {
      entry: path.resolve(__dirname, 'src/styles/ecl-button-ec.scss'),
      dest: path.resolve(outputFolder, 'styles/ecl-button-ec.css'),
      options: {
        banner,
        includePaths,
        sourceMap: false,
        minify: true,
      },
    },
    {
      entry: path.resolve(__dirname, 'src/styles/ecl-button-eu.scss'),
      dest: path.resolve(outputFolder, 'styles/ecl-button-eu.css'),
      options: {
        banner,
        includePaths,
        sourceMap: false,
        minify: true,
      },
    },
  ],
};
