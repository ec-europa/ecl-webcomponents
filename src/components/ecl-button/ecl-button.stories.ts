import iconsAllEc from '@ecl/resources-ec-icons/dist/lists/all.json';

const getArgs = () => {
  return {
    label: 'Button label',
    type: 'button',
    variant: 'primary',
    icon: '',
    iconPosition: 'after',
    iconTransform: '',
    hideLabel: false,
    indicator: false,
    indicatorValue: 3,
  };
};

const getArgTypes = () => {
  return {
    color_mode: { table: { disable: true } },
    variant: {
      name: 'button variant',
      type: { name: 'select' },
      options: ['primary', 'secondary', 'ghost', 'call'],
      description: "Button variant"
    },
    type: {
      name: 'button type',
      type: { name: 'select' },
      options: ['button', 'submit'],
      description: "Button type"
    },
    label: {
      name: 'button label',
      type: { name: 'string'},
      description: 'The text in the button',
    },
    hideLabel: {
      name: 'Icon only',
      type: { name: 'boolean'},
      description: 'Hides the label (not for screen-readers)',
      if: { arg: 'icon', truthy: true },
    },
    icon: {
      name: 'icon',
      type: { name: 'select' },
      options: iconsAllEc,
      description: "Add an icon",
    },
    iconPosition: {
      name: 'icon position',
      type: { name: 'select' },
      options: ['before', 'after'],
      description: 'Before or after the button label',
    },
    iconTransform: {
      name: 'icon transform',
      type: { name: 'select' },
      description: 'Button icon transform',
      options: [
        'rotate-90',
        'rotate-180',
        'rotate-270',
        'flip-horizontal',
        'flip-vertical',
      ],
    },
    indicator: {
      name: 'indicator',
      type: { name: 'boolean' },
      description: 'Show an indicator (only visible when label is hidden and icon exists)',
      if: { arg: 'icon', truthy: true },
    },
    indicatorValue: {
      name: 'indicator value',
      type: { name: 'number' },
      description: 'Value of the indicator (only visible when showIndicator is true)',
      if: { arg: 'showIndicator', truthy: false },
    },
  };
};

export default {
  title: 'Components/button',
};

const Template = (args) => {
  return `<ecl-button
    type="${args.type}"
    data-ecl-button-icon
    theme="${args.theme}"
    variant="${args.variant}"
    ${args.hideLabel ? 'hide-label="true"' : ''}s
    ${args.indicator && args.hideLabel ? `indicator indicator-value="${args.indicatorValue}"` : ''}
  >
    ${args.icon && args.iconPosition === 'before' ?
      `<ecl-icon 
        slot="icon-before"
        icon="${args.icon}"
        transform="${args.iconTransform}"
        theme="${args.theme}"
      >
      </ecl-icon>` : ''}
    ${args.label}
    ${args.icon && args.iconPosition === 'after' ?
      `<ecl-icon 
        slot="icon-after"
        icon="${args.icon}"
        theme="${args.theme}"
        transform="${args.iconTransform}"
      >
      </ecl-icon>` : ''}
  </ecl-button>`;
}

export const Button = Template.bind({});
Button.storyName = 'default';
Button.args = getArgs();
Button.argTypes = getArgTypes();
