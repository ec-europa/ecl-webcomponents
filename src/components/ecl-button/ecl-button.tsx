import { Component, Prop, h, Element } from '@stencil/core';

@Component({
  tag: 'ecl-button',
  styleUrls: {
    ec: './build/styles/ecl-button-ec.css',
    eu: './build/styles/ecl-button-eu.css',
  },
  shadow: false,
  scoped: true,
})

export class EclButton {
  @Element() el: HTMLElement;
  @Prop() styleClass: string = '';
  @Prop() type: string = 'primary';
  @Prop() theme: string = 'ec';

  componentDidRender() {
    if (this.el.getElementsByTagName('ecl-icon')[0] && this.el.querySelector('.ecl-icon')) {
      const slot = this.el.getElementsByTagName('ecl-icon')[0].getAttribute('slot');
      this.el.querySelector('.ecl-icon').classList.add('ecl-button__icon');
      if (slot) {
        this.el.querySelector('.ecl-icon').classList.add(`ecl-button__${slot.substring(0, 5) + '-' + slot.substring(5)}`);
      }
    }
  }

  getClass(): string {
    return [
      `ecl-button`,
      `ecl-button--${this.type}`,
      this.styleClass
    ].join(' ');
  }

  render() {
    return (
      <button class={this.getClass()}>
        <span class="ecl-button__container">
          <slot name="icon-before"></slot>
          <span class="ecl-button__label">
            <slot></slot>
          </span>
          <slot name="icon-after"></slot>
        </span>
      </button>
    )
  }
}
