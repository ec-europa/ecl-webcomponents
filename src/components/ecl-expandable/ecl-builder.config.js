const path = require('path');
const pkg = require('./package.json');

const nodeModules = path.resolve(__dirname, '../../../node_modules');
const outputFolder = './build';

// SCSS includePaths
const includePaths = [nodeModules];

const banner = `${pkg.name} - ${pkg.version} Built on ${new Date().toISOString()}`;

module.exports = {
  scripts: [
    {
      entry: path.resolve(nodeModules, '@ecl/vanilla-component-expandable/expandable.js'),
      dest: path.resolve(outputFolder, 'scripts/ecl-expandable-vanilla.js'),
      options: {
        banner,
        moduleName: 'ECL',
        includePaths,
        sourceMap: false,
      },
    },
  ],
  styles: [
    {
      entry: path.resolve(__dirname, 'src/styles/ecl-expandable-ec.scss'),
      dest: path.resolve(outputFolder, 'styles/ecl-expandable-ec.css'),
      options: {
        banner,
        includePaths,
        sourceMap: false,
        minify: true,
      },
    },
    {
      entry: path.resolve(__dirname, 'src/styles/ecl-expandable-eu.scss'),
      dest: path.resolve(outputFolder, 'styles/ecl-expandable-eu.css'),
      options: {
        banner,
        includePaths,
        sourceMap: false,
        minify: true,
      },
    },
  ],
};
