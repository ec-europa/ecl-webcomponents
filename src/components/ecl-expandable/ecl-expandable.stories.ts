export default {
  title: 'Components/expandable',
};

const Template = (args) =>  `<ecl-expandable
                              theme="${args.theme}"
                              id="${args.id}"
                              label-collapsed="${args.labelCollapsed}"
                              label-expanded="${args.labelExpanded}"
                              ecl-script
                            >
                              ${args.content}
                            </ecl-expandable>`;

export const Expandable = Template.bind({});
Expandable.storyName = 'default';
Expandable.args = {
  theme: 'ec',
  id: 'expandable-example',
  labelCollapsed: 'Expand',
  labelExpanded: 'Collapse',
  content: `<p class="ecl-u-type-paragraph-m">
    The EU is building an energy union that ensures Europeâ€™s energy supply is safe,
    viable and accessible to all. In doing so, it can boost the economy and attract investments that can create new jobs
    opportunities.
  </p>`,
};
Expandable.argTypes = {
  theme: {
    type: 'select',
    options: ['ec', 'eu'], 
    description: 'Ec or EU theme'
  },
  labelCollapsed: {
    name: 'label-collapsed',
  },
  labelExpanded: {
    name: 'label-expanded',
  },
  content: {
    name: 'content',
    description: 'Content inside the <ecl-expandable></expandable>',
  }
};