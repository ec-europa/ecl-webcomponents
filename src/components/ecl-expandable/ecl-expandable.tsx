import { Component, h, Prop, Element, getAssetPath } from '@stencil/core';
declare const ECL: any;

@Component({
  tag: 'ecl-expandable',
  styleUrls: {
    ec: './build/styles/ecl-expandable-ec.css',
    eu: './build/styles/ecl-expandable-eu.css',
  },
  shadow: false,
  scoped: true,
  assetsDirs: ['build'],
})

export class EclExpandable {
  @Element() el: HTMLElement;
  @Prop() styleClass: string = '';
  @Prop() eclScript: boolean = false;
  @Prop() withUtils: boolean = false;
  @Prop() isExpanded: boolean = false;
  @Prop() labelCollapsed: string = '';
  @Prop() labelExpanded: string = '';
  @Prop() theme: string = 'ec';

  componentWillLoad() {
    const src = getAssetPath('./build/scripts/ecl-expandable-vanilla.js');
    if (document.querySelector(`script[src="${src}"]`)) {
      document.querySelector(`script[src="${src}"]`).remove();
    }
    const script = document.createElement('script');
    script.src = src;
    script.onload = () => {
      const expandable = new ECL.Expandable(this.el);
      expandable.init();
    };
    document.body.appendChild(script);

    if (this.withUtils && !document.querySelector('#ecl-utils-styles')) {
      const style = document.createElement('style');
      style.id = 'ecl-utils-styles';
      style.innerHTML = `@import "./build/styles/ecl-utilities-${this.theme}.css"`;
      document.body.appendChild(style);
    }
  }

  getClass(): string {
    return [
      `ecl-expandable`,
      this.styleClass
    ].join(' ');
  }

  render() {
    return (
      <div
        class={this.getClass()}
      > 
        <ecl-button
          type="secondary"
          style-class="ecl-expandable__toggle"
          aria-controls={`${this.el.id}-content`}
          data-ecl-expandable-toggle
          data-ecl-label-expanded={this.labelExpanded}
          data-ecl-label-collapsed={this.labelCollapsed} 
        >

          <span
            class="ecl-button__label"
            data-ecl-label
          >
            {this.labelCollapsed}
          </span>
          <ecl-icon
            slot="icon-after"
            icon="corner-arrow"
            size="fluid"
            theme={this.theme}
            transform="rotate-180"
            style-class={`ecl-button__icon ecl-button__icon--after sc-ecl-button-${this.theme}`}
          ></ecl-icon>
        </ecl-button>
        <div
          class="ecl-expandable__content"
          id={`${this.el.id}-content`}
          hidden
        >
          <slot></slot>
        </div>
      </div>
    )
  }
}