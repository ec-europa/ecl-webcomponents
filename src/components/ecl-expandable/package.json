{
  "name": "@ecl/web-components-expandable",
  "author": "European Commission",
  "license": "EUPL-1.2",
  "version": "0.12.0",
  "description": "ECL Expandable",
  "main": "ecl-expandable.js",
  "private": true,
  "scripts": {
    "build:styles": "cross-env ecl-builder styles",
    "build:scripts": "cross-env ecl-builder scripts",
    "build": "run-s clean build:*",
    "clean": "rimraf build",
    "prepublish": "npm run build"
  },
  "devDependencies": {
    "@ecl/builder": "3.13.0",
    "@ecl/specs-component-expandable": "3.13.0",
    "@ecl/theme-ec": "3.13.0",
    "@ecl/theme-eu": "3.13.0",
    "@ecl/vanilla-component-button": "3.13.0",
    "@ecl/vanilla-component-expandable": "3.13.0",
    "cross-env": "7.0.3"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/planctus/ecl-webcomponents.git"
  },
  "bugs": {
    "url": "https://github.com/planctus/ecl-webcomponents/issues"
  },
  "homepage": "https://github.com/planctus/ecl-webcomponents",
  "keywords": [
    "ecl",
    "europa-component-library",
    "design-system",
    "web-components"
  ]
}
