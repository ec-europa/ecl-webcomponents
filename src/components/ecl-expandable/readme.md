# ecl-expandable

<!-- Auto Generated Below -->


## Properties

| Property         | Attribute         | Description | Type      | Default |
| ---------------- | ----------------- | ----------- | --------- | ------- |
| `eclScript`      | `ecl-script`      |             | `boolean` | `false` |
| `isExpanded`     | `is-expanded`     |             | `boolean` | `false` |
| `labelCollapsed` | `label-collapsed` |             | `string`  | `''`    |
| `labelExpanded`  | `label-expanded`  |             | `string`  | `''`    |
| `styleClass`     | `style-class`     |             | `string`  | `''`    |
| `theme`          | `theme`           |             | `string`  | `'ec'`  |
| `withUtils`      | `with-utils`      |             | `boolean` | `false` |


## Dependencies

### Depends on

- [ecl-button](../ecl-button)
- [ecl-icon](../ecl-icon)

### Graph
```mermaid
graph TD;
  ecl-expandable --> ecl-button
  ecl-expandable --> ecl-icon
  style ecl-expandable fill:#f9f,stroke:#333,stroke-width:4px
```

----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
