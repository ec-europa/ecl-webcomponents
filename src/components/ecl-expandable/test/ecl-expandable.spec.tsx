import { newSpecPage } from '@stencil/core/testing';
import { EclExpandable } from '../ecl-expandable';
import { EclButton } from '@ecl/web-components-button/ecl-button';

describe('ecl-expandable', () => {
  it('renders', async () => {
    const page = await newSpecPage({
      components: [EclExpandable, EclButton],
      html: `<ecl-expandable
              theme="ec"
              el-id="expandable-id"
              icon-path="/icons.svg"
              label-collapsed="Expand"
              label-expanded="Collapse"
              ecl-script
            >
              <p class="ecl-u-type-paragraph-m">
                The EU is building an energy union that ensures Europeâ€™s energy supply is safe,
                viable and accessible to all. In doing so, it can boost the economy and attract investments that can create new jobs
                opportunities.
              </p>
            </ecl-expandable>`,
    });

    expect(page.root).toMatchSnapshot();
  });
});