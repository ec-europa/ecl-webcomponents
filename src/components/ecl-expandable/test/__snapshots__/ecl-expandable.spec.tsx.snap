// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ecl-expandable renders 1`] = `
<ecl-expandable ecl-script="" el-id="expandable-id" icon-path="/icons.svg" label-collapsed="Expand" label-expanded="Collapse" theme="ec">
  <!---->
  <div class="ecl-expandable">
    <ecl-button data-ecl-expandable-toggle data-ecl-label-collapsed="Expand" data-ecl-label-expanded="Collapse" style-class="ecl-expandable__toggle sc-ecl-expandable-ec">
      <!---->
      <button aria-controls="expandable-id-content" aria-expanded="false" class="ecl-button ecl-button--ghost ecl-expandable__toggle sc-ecl-expandable-ec" data-ecl-expandable-toggle="eclExpandableToggle" data-ecl-label-collapsed="Expand" data-ecl-label-expanded="Collapse" type="button">
        <span class="ecl-button__container">
          <span class="ecl-button__label">
            <span class="ecl-button__label" data-ecl-label>
              Expand
            </span>
          </span>
          <ecl-icon icon="corner-arrow" size="fluid" slot="icon-after" style-class="ecl-button__icon ecl-button__icon--after sc-ecl-button-ec sc-ecl-expandable-ec" theme="ec" transform="rotate-180"></ecl-icon>
        </span>
      </button>
    </ecl-button>
    <div class="ecl-expandable__content" hidden="" id="expandable-id-content">
      <p class="ecl-u-type-paragraph-m sc-ecl-expandable-ec">
        The EU is building an energy union that ensures Europeâ€™s energy supply is safe, viable and accessible to all. In doing so, it can boost the economy and attract investments that can create new jobs opportunities.
      </p>
    </div>
  </div>
</ecl-expandable>
`;
